### PR TITLE
Added event_stop method for disconnecting mqtt.

### PR DESCRIPTION
**Pull request by rebinmannady93**
# Description of change

Event_Stop Method

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Tested in Example.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/client/api/events/node_event.h [hunk 1]
```diff
@@ -113,6 +113,14 @@ int event_unsubscribe(event_client_handle_t client, int *mid, char *topic);
  */
 int event_start(event_client_handle_t client);
 
+/**
+ * @brief Mqtt disconnect
```
Comment (oopsmonk) [inline src/client/api/events/node_event.h]: please give more info about what this method does, like
```suggestion
 * @brief Disconnect from the MQTT broker
```

File: src/client/api/events/node_event.h [hunk 2]
```diff
@@ -113,6 +113,14 @@ int event_unsubscribe(event_client_handle_t client, int *mid, char *topic);
...
+ *
+ * @param[in] client Event client instance
+ * @return 0 if Success
+ */
+int event_stop(event_client_handle_t client);
+
 /**
  * @brief Destroy event intance
```
Comment (oopsmonk) [inline src/client/api/events/node_event.h]:
```suggestion
 * @brief Stop and destroy event instance
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: src/client/api/events/node_event.h (hunk 1 above)
Comment (rebinmannady93) [reply src/client/api/events/node_event.h]: Done.

File: src/client/api/events/node_event.h (hunk 2 above)
Comment (rebinmannady93) [reply src/client/api/events/node_event.h]: Done.

**Pull request merged**